Windows Plugins: Add new parameter to check_disk to show used space

mcktr opened the pull request:
### Feature
The Linux `disk` check returns the used space from a given or all disks. I thought it would be a good feature to have the possibility on windows also. 

This adds a new parameter to the check_disk windows plugin. With `--show-used` or `-U` this plugin will output the used space instead of the free space. 

To get the used space the plugin calculate the total space minus the free space of a disk.

### Test environment:
Windows 10 Pro x64
Disk G:
* capacity 5,65 GB
* used space 1,03 GB
* free space 4,62 GB

#### Testing the free space output:

```
check_disk.exe -u gb -p G:
DISK OK - free space:G:\ 5 GB (83%); | G:\=5GB;;;0;6

check_disk.exe -u gb -p G: -w 10% -c 5%
DISK OK - free space:G:\ 5 GB (83%); | G:\=5GB;0.60000000000000009;0.30000000000000004;0;6

check_disk.exe -u gb -p G: -w 90% -c 5%
DISK WARNING - free space:G:\ 5 GB (83%); | G:\=5GB;5.4000000000000004;0.30000000000000004;0;6

check_disk.exe -u gb -p G: -w 90% -c 95%
DISK CRITICAL - free space:G:\ 5 GB (83%); | G:\=5GB;5.4000000000000004;5.6999999999999993;0;6

```
#### Testing the used space output:
```

check_disk.exe -u gb -p G: -U
DISK CRITICAL - used space:G:\ 1 GB (17%); | G:\=1GB;;;0;6

check_disk.exe -u gb -p G: -U -w 90% -c 95%
DISK OK - used space:G:\ 1 GB (17%); | G:\=1GB;5.4000000000000004;5.6999999999999993;0;6

check_disk.exe -u gb -p G: -U -w 10% -c 95%
DISK WARNING - used space:G:\ 1 GB (17%); | G:\=1GB;0.60000000000000009;5.6999999999999993;0;6

check_disk.exe -u gb -p G: -U -w 10% -c 11%
DISK CRITICAL - used space:G:\ 1 GB (17%); | G:\=1GB;0.60000000000000009;0.66000000000000003;0;6
```

As far as I can tell, this should not break any things of previous versions.

Crunsher commented:
Not too happy about having two extra function for this change. Especially the print functions are 90% the same.

Could you merge the getFreeAndCap/getUsedAndCap and printOutputFree/printOutputUsed or explain why you choose to add them?

mcktr commented:
I used extra methods for it because I thought this were the best way to make sure I don't break something that is already there. Maybe this was the wrong approach.

I refactored the code and merged the functions together. Also I found a misbehavior with the returning stats, as you can see in my tests above that the output is critical if I don't set any thresholds - I changed this, so the output should be ok.

I used the same test environment as above

#### testing the free space
```
check_disk.exe -u gb -p G:
DISK OK - free space:G:\ 5 GB (83%); | G:\=5GB;;;0;6

check_disk.exe -u gb -p G: -w 10% -c 5%
DISK OK - free space:G:\ 5 GB (83%); | G:\=5GB;0.60000000000000009;0.30000000000000004;0;6

check_disk.exe -u gb -p G: -w 90% -c 5%
DISK WARNING - free space:G:\ 5 GB (83%); | G:\=5GB;5.4000000000000004;0.30000000000000004;0;6

check_disk.exe -u gb -p G: -w 90% -c 95%
DISK CRITICAL - free space:G:\ 5 GB (83%); | G:\=5GB;5.4000000000000004;5.6999999999999993;0;6
```

#### testing the used space
```
check_disk.exe -u gb -p G: -U
DISK OK - used space:G:\ 1 GB (17%); | G:\=1GB;;;0;6

check_disk.exe -u gb -p G: -U -w 90% -c 95%
DISK OK - used space:G:\ 1 GB (17%); | G:\=1GB;5.4000000000000004;5.6999999999999993;0;6

check_disk.exe -u gb -p G: -U -w 10% -c 95%
DISK WARNING - used space:G:\ 1 GB (17%); | G:\=1GB;0.60000000000000009;5.6999999999999993;0;6

check_disk.exe -u gb -p G: -U -w 10% -c 11%
DISK CRITICAL - used space:G:\ 1 GB (17%); | G:\=1GB;0.60000000000000009;0.66000000000000003;0;6
```

Crunsher commented:
I re-factored the code a bit to conform to our style and included the new parameter in the ITL and documentation.

Will merge once I tested a few edge cases
